### PR TITLE
Improves `Popen` types in `psutil`

### PR DESCRIPTION
**Pull request by sobolevn**
1. `__dir__` was removed, because it has regular return type from `object`
2. `Popen.wait` has the same type as `Process.wait`

### CODE REVIEW

File: stubs/psutil/psutil/__init__.pyi [hunk 1]
```diff
@@ -160,13 +160,13 @@ class Process:
     def kill(self) -> None: ...
     def wait(self, timeout: int | None = ...) -> int: ...
 
+_Popen = TypeVar("_Popen", bound="Popen")
```
Comment (srittau) [inline stubs/psutil/psutil/__init__.pyi:163]: For future reference: Use `_typeshed.Self` in cases like this. Using a bound type for self references is unnecessary.

Comment (sobolevn) [reply stubs/psutil/psutil/__init__.pyi:163]: Thanks! 👍 

I will change this later.

**Comment by Akuli** [comment]
Perhaps we should just alias this to `subprocess.Popen`? It appears to be basically `subprocess.Popen` with a bug fix, and for typing purposes there shouldn't be any difference.

**Comment by sobolevn** [comment]
@Akuli great idea. I've opened a PR based on your suggestion https://github.com/python/typeshed/pull/6078

**Pull request merged**